client/src/components/Login/Login.js

```diff
@@ -8,26 +8,19 @@ const Login = () => {
   const { auth, login } = useAuth()
   const navigate = useNavigate()
   let location = useLocation()
-
-  // Assignment: redirect the newly logged in user to the page they were on
-  // OR to the User component
-
-  const source = location.state?.from?.pathname || "/User"
-  const backTologin = location.state?.from?.pathname || "/Login"
+  const source = location.state?.from?.pathname || "/app"
```

Comment:
You were closer before you reverted this. The goal was to redirect the user to the previous path or '/user'